src/v2/codegen/state.rs

```diff
@@ -425,9 +431,10 @@ impl EmitterState {
     }
 
     /// Validates crate metadata, sets the unset fields and returns a reference.
-    fn infer_crate_meta(&self) -> Result<Option<Rc<RefCell<CrateMeta>>>, Error> {
-        if let Some(m) = self.crate_meta.as_ref() {
-            let mut meta = m.borrow_mut();
+    fn infer_crate_meta(&self) -> Result<Option<Rc<RefCell<Option<CrateMeta>>>>, Error> {
```

Comment:
This function should return `Result<Rc<RefCell<Option<CrateMeta>>>, Error>` - is there any particular reason for using `Result<Option<...>, Error>`?